stats/src/main/java/com/facebook/airlift/stats/cardinality/BucketListener.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.stats.cardinality;
+
+public interface BucketListener
```

Comment:
Note: The `BucketListener` pattern was taken from [upstream](https://github.com/airlift/airlift/blob/master/stats/src/main/java/io/airlift/stats/cardinality/BucketListener.java).